src/freemindlatex/__main__.py

```diff
@@ -156,12 +219,23 @@ def CompileDir(directory):
     _CompileInWorkingDirectory(work_dir)
 
     shutil.copyfile(
-      os.path.join(
-        work_dir, "slides.pdf"), os.path.join(
-          directory, "slides.pdf"))
+      os.path.join(work_dir, "slides.pdf"), target_pdf_loc)
+
     return True
 
   except LatexCompilationError as e:
+    shutil.copyfile(
+      os.path.join(work_dir, "slides.pdf"), target_pdf_loc)
+    latex_log_file = os.path.join(
+      directory, gflags.FLAGS.latex_error_log_filename)
+    with open(latex_log_file, 'w') as ofile:
+      ofile.write(str(e))
+    return False
+
+  except UnrecoverableLatexCompilationError as e:
```

Comment:
When not recoverable, should copy a file over, saying "unrecoverable", and the error messages. Even this is better than just showing the old page.